File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline

n = int(input())

stack = []
result = []
num_idx = 1
ok = 1
for i in range(n):
    num = int(input())
    while num_idx <= num:
        stack.append(num_idx)
        num_idx += 1
        result.append('+')
    if stack[-1] == num:
        stack.pop()
        result.append('-')
    else:
        ok = 0

if ok == 0:
    print('NO')
else:
    for r in result:
        print(r)

File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        int idx = 1;
        boolean ok = true;
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < n; i++) {
            int num = Integer.parseInt(br.readLine());

            while (idx <= num) {
                stack.push(idx++);
                sb.append("+\n");
            }

            if (stack.peek() == num) {
                stack.pop();
                sb.append("-\n");
            } else {
                ok = false;
            }
        }

        if (ok) {
            if (sb.length() > 0 && sb.charAt(sb.length() - 1) == '\n') {
                sb.deleteCharAt(sb.length() - 1);
            }
            System.out.print(sb.toString());
        } else {
            System.out.println("NO");
        }
    }
}
